smac/intensification/abstract_racer.py

```diff
@@ -117,43 +117,75 @@ def __init__(self,
-                        run_history: RunHistory,
-                        time_bound: float = float(MAXINT),
-                        log_traj: bool = True) -> typing.Tuple[Configuration, float]:
-        """
-        Runs intensification by evaluating one configuration-instance at a time
+    def eval_challenger(
+        self,
+        run_info: RunInfo,
+        time_bound: float = float(MAXINT),
+    ) -> typing.Tuple[StatusType, typing.Any, float, dict]:
+        """Runs a configuration with the provided input.
         *Side effect:* adds runs to run_history
 
         Parameters
         ----------
-        challenger : Configuration
-            promising configuration
-        incumbent : typing.Optional[Configuration]
-            best configuration so far, None in 1st run
-        run_history : smac.runhistory.runhistory.RunHistory
-            stores all runs we ran so far
+        run_info : RunInfo
+            An object that specifies the inputs to a tae runner
         time_bound : float, optional (default=2 ** 31 - 1)
             time in [sec] available to perform intensify
-        log_traj : bool
-            whether to log changes of incumbents in trajectory
 
         Returns
         -------
-        typing.Tuple[Configuration, float]
-            incumbent and incumbent cost
+        status: StatusType
+            The status of the execution of a given config
+            {SUCCESS, TIMEOUT/CAPPED, CRASHED, ABORT}
+        cost: typing.Any
+            cost/regret/quality (typing.Any)
+        duration: float
+            runtime (None if not returned by TA)
+        res: Dict
+            additional info
         """
 
-        raise NotImplementedError()
+        if time_bound < self._min_time:
+            raise ValueError("time_bound must be >= %f" % self._min_time)
+
+        if run_info.config is None or run_info.config is None:
+            raise ValueError("Call to eval configurations without any valid config")
+
+        try:
+            # Cap makes sense only in challenger. For incumbent runs, capped will be
+            # false. For a challenger, the run_info.cutoff will be smaller than self.cutoff
+            capped = False
+            if (self.cutoff is not None) and (run_info.cutoff < self.cutoff):
+                capped = True
+
+            status, cost, dur, res = self.tae_runner.start(
+                config=run_info.config,
+                instance=run_info.instance,
+                seed=run_info.seed,
+                cutoff=run_info.cutoff,
+                budget=run_info.budget,
+                instance_specific=self.instance_specifics.get(run_info.instance, "0"),
+                capped=capped
+            )
+
+        except CappedRunException:
+            self.logger.debug("Challenger itensification timed out due "
+                              "to adaptive capping.")
+            status, cost, dur, res = StatusType.CAPPED, float(MAXINT), 0.0, {}
```

Comment:
isn't it MAXINT? because that is defined to be the cost for a crash:
smac/tae/execute_func.py:        cost_for_crash: float = float(MAXINT)

Thanks, let me change the 0.0 to be the run_info.cutoff, as it is more informative

Reply:
No, I think it should be like this https://github.com/automl/SMAC3/blob/d5b93811aa74e0513e580576ebb0fbf857e18f1a/smac/tae/execute_ta_run.py#L239